### DISCUSSION TITLE
Help retrieving my generated data

### DISCUSSION THREAD

**Discussion by mockinbirdy** [LostRuins/koboldcpp#1288] [Q&A]
Hi there, I’m using koboldcpp. I urgently need help recovering tabs with important generated information.

Here’s what happened: I had 4 browser tabs open, each running a separate koboldcpp chat on localhost. These were active chats I'd been working on for several days. Unfortunately, my power went out, and when my PC rebooted, everything was gone.

When I reopened my browser and accessed just one tab, it displayed the most recent saved generations (which was no surprise to me, and that was the reason I kept several tabs running due to koboldcpp not being able to conveniently keep several chats going in the same browser at the same time). However, I haven’t touched the other 3 tabs yet, and I’m worried about losing their data if I open them and they refresh.

I’m using brave on windows. Is there a way to safely recover the other 3 tabs without risking the information? 

Thank you in advance, and Merry Christmas 🎄

**Comment by LostRuins** [comment]
Hi there. This is unfortunate. Unfortunately, Lite only maintains one autosaved session at any time - running a separate session in another window on the same browser will cause your autosave data to be overwritten!

Why not use the **save slots or JSON savefile downloads** to preserve the data? That's the optimal way to ensure that your session doesn't get lost.

![image](https://github.com/user-attachments/assets/10f748d6-0c2b-48a2-8323-2c413ad00093)

Nevertheless, now that your browser has restarted it's likely that the other sessions have already been overwritten by the latest one. If your browser somehow kept saved form data, there *might* be a chance to recover them by opening each window and exporting it out to a JSON save on disk. However, this is not very likely, if the window is refreshed, it will simply only have the contents of the latest autosave.

If you must rely on the autosave feature alone, you can have multiple sessions by using different browsers. For example, installing Edge, Chrome and Firefox will allow you to run 3 persistent sessions at the same time.

I would still strongly recommend either using the save slots or exporting to JSON savefile - that's the best way to ensure the data does not get overwritten.

**Reply by mockinbirdy** [reply]
Hi, thanks for your reply. It’s also not very convenient because I need to use different parameters in different tabs (and changing them every half an hour is definitely not comfortable). Is there hope this problem will be addressed in future releases? Maybe so the user can open several concurrent tabs running simultaneously?

And while we're at it, is there a way to know what kind of settings are used under the hood of `chat mode` in the usage mode settings in kobold? I don't seem to see a way to replicate it in tavern, for example.

**Reply by LostRuins** [reply]
The parameters can be bound to each of the save slots. So you can set each slot up with the generation settings that you like for it.

The other option is also using a second browser.

To see the settings used, you can take a look at the koboldcpp terminal after generating.

**Comment by mockinbirdy** [comment]
I was also hoping to hear advice from brave users. Maybe it’s still possible to retrieve the info from the previous sessions by accessing `localstorage` files or something similar...? If yes, how would I go about it?
